fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/impl/registry/sync/RegistrySyncManager.java

```diff
@@ -216,4 +268,41 @@ public static void unmap() throws RemapException {
 			}
 		}
 	}
+
+	public static void bootstrapRegistries() {
+		for (MutableRegistry<?> registry : Registry.REGISTRIES) {
+			if (registry instanceof HashedRegistry) {
+				((HashedRegistry) registry).storeHash();
+			}
+		}
+
+		postBootstrap = true;
+	}
+
+	// Checks the stored hash against the current hash, if it has changed mark as modded
+	public static void checkRegistryHashes() {
+		for (MutableRegistry<?> registry : Registry.REGISTRIES) {
+			if (registry instanceof HashedRegistry) {
+				if (((HashedRegistry) registry).getStoredHash() != ((HashedRegistry) registry).storeHash()) {
+					RegistryAttributeHolder<?> holder = RegistryAttributeHolder.get(registry);
+
+					if (!holder.hasAttribute(RegistryAttribute.MODDED)) {
+						LOGGER.debug("Registry {} has been marked as modded as the hash changed since bootstrap", Registry.REGISTRIES.getId(registry));
+						holder.addAttribute(RegistryAttribute.MODDED);
+					}
+				}
+			}
+		}
+	}
+
+	public static boolean isRegistryModded(Identifier registryId) {
+		//All none minecraft registries are modded
+		if (!registryId.getNamespace().equals("minecraft")) {
+			return true;
+		}
```

Comment:
This should prob be moved from here, to adding `RegistryAttribute.MODDED` when creating the RegistryAttributeHolder